pkg/v1/remote/write.go

```diff
@@ -57,9 +57,25 @@ func Write(ref name.Reference, img v1.Image, auth authn.Authenticator, t http.Ro
 	}
 
 	// Upload individual layers in goroutines and collect any errors.
+	// If we can dedupe by the layer digest, try to do so. If the layer is
+	// a stream.Layer, we can't dedupe and might re-upload.
 	var g errgroup.Group
+	seen := map[v1.Hash]struct{}{}
```

Comment:
If you call this uploaded - it'll probably be a little clearer.

Also if you use map[v1.Hash]bool, then you can do

if uploaded[h]